### PR TITLE
Bump razor

### PR DESCRIPTION
**Pull request by ryzngard**
Release from build https://dev.azure.com/dnceng/internal/_build/results?buildId=2414428&view=results

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

**Review by dibarbet — APPROVED** [review]
Would you mind updating the changelog as well with the fixes in this insertion?

**Comment by ryzngard** [comment]
> Would you mind updating the changelog as well with the fixes in this insertion?

I did it manually by diffing builds, but is there a tool we have for this?

**Comment by dibarbet** [comment]
> I did it manually by diffing builds, but is there a tool we have for this?

In the razor repo, you can run `roslyn-tools pr-finder -p <previous commit> -c <current commit> --format o#` once you have the [roslyn-tools global tool ](https://github.com/dotnet/roslyn-tools/tree/main/src/dotnet-roslyn-tools)installed.

Then I usually filter out everything that isn't relevant for vscode / adjust titles to better explain the changes.

**Comment by davidwengier** [comment]
I knew about the `pr-finder`, and used to use it weekly for VS Mac insertions, and I have written every single changelog update manually. I cannot believe I didn't think of this. 🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@
...
 
 ## Latest
+* Bump razor and update changelog (PR: [#7005](https://github.com/dotnet/vscode-csharp/pull/7005))
+  * Tweak RazorConfigurationFormatter to account for serialization format change (PR: [#10157](https://github.com/dotnet/razor/pull/10157))
+  * Remove lock in BackgroundDocumentGenerator (PR: [#10159](https://github.com/dotnet/razor/pull/10157))
+  * Use PooledArrayBuilder<SyntaxToken> throughout parsers and tokenizer (PR [#10095](https://github.com/dotnet/razor/pull/10095))
+  * Fix hot path string allocations from ProjectKey (PR: [#10138](https://github.com/dotnet/razor/pull/10138))
+  * Disable latest runtime patch (PR: [#10133](https://github.com/dotnet/razor/pull/10133))
+
+
```
Comment (davidwengier) [inline CHANGELOG.md:14]: Nits: Two blank lines (though I doubt markdown cares) and BackgroundDocumentGenerator is not used in VS Code (though I doubt anyone reading the notes cares.. or indeed reads the notes 😛)

Comment (ryzngard) [reply CHANGELOG.md:14]: SpaceballsFunnyGIF

**Comment by maryamariyan** [comment]
Thanks for the comments and thank you @ryzngard for helping out

**Pull request merged**